link/src/test/java/com/stripe/android/link/account/DefaultLinkAccountManagerTest.kt

```diff
@@ -145,338 +131,596 @@ class DefaultLinkAccountManagerTest {
+                override suspend fun consumerSignUp(
+                    email: String,
+                    phone: String,
+                    country: String,
+                    name: String?,
+                    consentAction: ConsumerSignUpConsentAction
+                ): Result<ConsumerSessionSignup> {
+                    if (consentAction == ConsumerSignUpConsentAction.CheckboxWithPrefilledEmailAndPhone) {
+                        callCount += 1
+                    }
+                    return super.consumerSignUp(email, phone, country, name, consentAction)
+                }
+            }
+            accountManager(linkRepository = linkRepository).signInWithUserInput(
                 createUserInputWithAction(
                     SignUpConsentAction.CheckboxWithPrefilledEmailAndPhone
                 )
             )
 
-            verifyConsumerAction(ConsumerSignUpConsentAction.CheckboxWithPrefilledEmailAndPhone)
+            assertThat(linkRepository.callCount).isEqualTo(1)
         }
 
     @Test
     fun `signInWithUserInput sends correct consumer action on 'Implied' consent action`() = runSuspendTest {
-        accountManager().signInWithUserInput(createUserInputWithAction(SignUpConsentAction.Implied))
+        val linkRepository = object : FakeLinkRepository() {
+            var callCount = 0
+            override suspend fun consumerSignUp(
+                email: String,
+                phone: String,
+                country: String,
+                name: String?,
+                consentAction: ConsumerSignUpConsentAction
+            ): Result<ConsumerSessionSignup> {
+                if (consentAction == ConsumerSignUpConsentAction.Implied) {
+                    callCount += 1
+                }
+                return super.consumerSignUp(email, phone, country, name, consentAction)
+            }
+        }
+        accountManager(linkRepository = linkRepository).signInWithUserInput(
+            createUserInputWithAction(
+                SignUpConsentAction.Implied
+            )
+        )
 
-        verifyConsumerAction(ConsumerSignUpConsentAction.Implied)
+        assertThat(linkRepository.callCount).isEqualTo(1)
     }
 
     @Test
     fun `signInWithUserInput sends correct consumer action on 'ImpliedWithPrefilledEmail' consent action`() =
         runSuspendTest {
-            accountManager().signInWithUserInput(
+            val linkRepository = object : FakeLinkRepository() {
+                var callCount = 0
+                override suspend fun consumerSignUp(
+                    email: String,
+                    phone: String,
+                    country: String,
+                    name: String?,
+                    consentAction: ConsumerSignUpConsentAction
+                ): Result<ConsumerSessionSignup> {
+                    if (consentAction == ConsumerSignUpConsentAction.ImpliedWithPrefilledEmail) {
+                        callCount += 1
+                    }
```

Comment:
not necessary to update, but just as some feedback. I think it's more idiomatic to save the consentAction that's passed in here and then verify that it matches the argument you expect, rather than tracking the call count.

It seems more likely that this test would fail because we'd call this function with the wrong params and not that it would fail because we're calling it too frequently. Checking the param type would lead to an easier to debug failure in the case that the test failed because the param was incorrect